### PR TITLE
feat: add example-test for rating

### PR DESCRIPTION
**Pull request by 0xff-dev**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it
feat: add example-test for rating

#### Which issue(s) this PR fixes

Fix: https://github.com/kubebb/core/issues/293

#### Special notes for your reviewer
https://github.com/0xff-dev/core/actions/runs/5973617083/job/16206305101

### CODE REVIEW

File: config/samples/tekton_test.sh [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+function info() {
+	cecho -c 'blue' "$@"
+}
+
+function waitComponentStatus() {
+	namespace=$1
+	componentName=$2
+	START_TIME=$(date +%s)
+	while true; do
+		versions=$(kubectl -n${namespace} get components.core.kubebb.k8s.com.cn ${componentName} -ojson | jq -r '.status.versions|length')
```
Comment (Abirdcfly) [inline config/samples/tekton_test.sh:131]: --ignore-not-found=true

File: config/samples/tekton_test.sh [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+		if [[ $versions -ne 0 ]]; then
+			echo "component ${componentName} already have version information and can be installed"
+			#			digest=$(kubectl -n${namespace} get components.core.kubebb.k8s.com.cn ${componentName} -ojson | jq -r '.status.versions.digest')
```
Comment (Abirdcfly) [inline config/samples/tekton_test.sh:134]: remove this comment

File: config/samples/tekton_test.sh [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+	done
+}
+
+function waitRatingDone() {
+	namespace=$1
+	ratingName=$2
+	START_TIME=$(date +%s)
+	sleep 2 # wait for operator patch status. avoid 0=0 situations
+	while true; do
+		complete=$(kubectl -n${namespace} get rating ${ratingName} -ojson | jq '.status.pipelineRuns' | jq '{l:length,o:map(select(.conditions[0].type=="Succeeded" and .conditions[0].status=="True"))|length}' | jq '.l == .o')
```
Comment (Abirdcfly) [inline config/samples/tekton_test.sh:154]: --ignore-not-found=true

File: config/samples/tekton_test.sh [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+		sleep 5
+	done
+}
+
+function checkCm() {
+	namespace=$1
+	cmname=$2
+	START_TIME=$(date +%s)
+	while true; do
+		cm=$(kubectl -n${namespace} get cm ${cmname} -ojson | jq -r '.metadata.name')
```
Comment (Abirdcfly) [inline config/samples/tekton_test.sh:174]: --ignore-not-found=true

**Review by Abirdcfly — APPROVED** [review]

File: config/samples/tekton_test.sh [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+
+KindName="kubebb-core"
+TimeoutSeconds=${TimeoutSeconds:-"300"}
+HelmTimeout=${HelmTimeout:-"1800s"}
+KindVersion=${KindVersion:-"v1.24.4"}
+TempFilePath=${TempFilePath:-"/tmp/kubebb-core-tekton-test"}
+KindConfigPath=${TempFilePath}/kind-config.yaml
+InstallDirPath=${TempFilePath}/building-base
+DefaultPassWord=${DefaultPassWord:-'passw0rd'}
+LOG_DIR=${LOG_DIR:-"/tmp/kubebb-core-example-test/logs"}
```
Comment (Abirdcfly) [inline config/samples/tekton_test.sh:34]: kubebb-core-tekton-test

**Review by Abirdcfly — APPROVED** [review]

File: .github/workflows/example_test.yaml [hunk 6]
```diff
@@ -45,6 +45,46 @@ jobs:
           key: operator.image-${{ github.sha }}
           path: /tmp/operator.image.tar
 
+  tekton-test:
```
Comment (bjwswang) [inline .github/workflows/example_test.yaml:48]: `tekton-test` ->  `rating-test`

File: config/samples/tekton_test.sh [hunk 7]
```diff
@@ -0,0 +1,286 @@
+#!/bin/bash
```
Comment (bjwswang) [inline config/samples/tekton_test.sh:1]: tekton-test -> rating-test

**Review by bjwswang — CHANGES_REQUESTED** [review]
tekton -> rating

**Review by bjwswang — APPROVED** [review]

**Pull request merged**